site/app/libraries/FileUtils.php

```diff
@@ -566,4 +566,77 @@ public static function areWordsInFile(string $file, array $words)
 
         return $words_detected;
     }
+
+    /**
+     * Given an array of uploaded files, makes sure they are properlly uploaded
+     *
+     * @param array $files - should be in the same format as the $_FILES[] variable
+     * @return array representing the status of each file
+     * e.g. array('name' => 'foo.txt','type' => 'application/octet-stream', 'error' =>      
+     *            'success','size' => 100, 'success' => true)
+     * if $files is null returns failed => no files sent to validate
+     */
+    public static function validateUploadedFiles($files){
+
+        if(!isset($files)){
+           return array("failed" => "No files sent to validate");
+        }
+
+        $ret = array();
+        $num_files = count($files['name']);
+        $max_size = Utils::returnBytes(ini_get('upload_max_filesize'));
+
+        for ($i = 0; $i < $num_files; $i++) {
+            //extract the values from each file
+            $name = $files['name'][$i];
+            $type = FileUtils::getMimeType($files['tmp_name'][$i]);
+            $size = $files['size'][$i];
+            $err_msg = "";
+            //did anything go wrong?
+            switch ($files['error'][$i]) {
+                case UPLOAD_ERR_OK:
+                    $err_msg = "success";
+                    break;
+                case UPLOAD_ERR_NO_FILE:
+                    $err_msg = "No file submitted";
+                    break;
+                case UPLOAD_ERR_INI_SIZE:
+                case UPLOAD_ERR_FORM_SIZE:
+                   $err_msg = "File \"" . $name . "\" too large got (" . Utils::formatBytes("mb", $size) . ")";
+                   break;
+                case UPLOAD_ERR_PARTIAL:
+                    $err_msg = "The uploaded file was only partially uploaded";
+                    break; 
+                case UPLOAD_ERR_NO_TMP_DIR:
+                    $err_msg = "Missing a temporary folder";
+                    break; 
+                case UPLOAD_ERR_CANT_WRITE:
+                    $err_msg = "Failed to write file to disk";
+                    break; 
+                case UPLOAD_ERR_EXTENSION:
+                    $err_msg = "File upload stopped by extension";
+                    break; 
+                default:
+                    $err_msg = "Unknown error";
+                    $success = false;
+            }
```

Comment:
Can you use [ErrorMessages::uploadErrors](https://github.com/Submitty/Submitty/blob/master/site/app/libraries/ErrorMessages.php#L20) function for this switch case. I think it's just missing the `UPLOAD_ERR_INI_SIZE` and `UPLOAD_ERR_FORM_SIZE` cases.

Reply:
Thanks for pointing that out!